modules/theme/script.js

```diff
@@ -0,0 +1,23 @@
+// Add dark / light detection that runs before Nuxt.js load.
```

Comment:
- As proposed in Notion, we can minify this injected script into a tiny IIFE
- `getCookie` can be reused by the plugin to save some bytes, mem and also more consistency (like `__nuxt_theme._getCookie`)
- Also other logic like query matching and setAttr is duplicated (not a big deal but still perf improvement area)